editoast/src/views/timetable/similar_trains.rs

```diff
@@ -318,7 +322,13 @@ async fn similar_trains(
         let end = segment.end().clone();
         #[cfg(debug_assertions)]
         std::fs::write("/tmp/dot.txt", graph.to_dot()).unwrap();
-        let mut state = graph::MatchingState::new(segment, graph);
+        let mut state = match graph::MatchingState::try_new(segment, graph) {
+            Ok(state) => state,
+            Err(_) => {
```

Comment:
```suggestion
            Err(()) => {
```

nit: doing so will trigger an error for us to fix if we change the Err type